src/main/scala/org/broadinstitute/dsde/firecloud/service/TrialService.scala

```diff
@@ -170,7 +170,27 @@ final class TrialService
                                       transitionPostProcessing: (Attempt, Option[UserTrialStatus], UserTrialStatus) => Unit): Future[PerRequestMessage] = {
     var results: Seq[(String, String)] = Seq()
     userEmails.foreach { userEmail =>
-      val finalStatus = samDao.adminGetUserByEmail(RawlsUserEmail(userEmail)) flatMap { regInfo =>
+      val finalStatus = executeStateTransition(managerInfo, userEmail, statusTransition, transitionPostProcessing)
+      // Use await here so that multiple Futures don't have a conflict when claiming a billing project
+
+      val result =
+        try {
+          Await.result(finalStatus, scala.concurrent.duration.Duration.Inf)
+        } catch {
+          case ex: FireCloudException =>
+            StatusUpdate.toName(StatusUpdate.ServerError(ex.getMessage))
+        }
```

Comment:
yeah, I really don't like the use of `Await` here, especially with a `Duration.Inf`. Step 1 is to reduce the duration to something that won't hang the server in an extreme case, e.g. `1.minute` ... step 2 will be to eliminate the blocking code entirely. I can take a shot at refactoring this too.